app/src/main/java/com/samco/trackandgraph/graphstatview/factories/LineGraphDataFactory.kt

```diff
@@ -163,6 +169,34 @@ class LineGraphDataFactory : ViewDataFactory<LineGraphWithFeatures, ILineGraphVi
         return if (plottingData.dataPoints.size >= 2) plottingData else null
     }
 
+
+    private suspend fun getYAxisParameters(
+            lineGraph: LineGraphWithFeatures,
+            series: Collection<FastXYSeries?>,
+            timeBasedRange: Boolean ) : Pair< RectRegion, Pair<StepMode, Double> > {
+        val fixed = lineGraph.yRangeType == YRangeType.FIXED;
+
+        val bounds = RectRegion()
+        series.forEach { it?.let { bounds.union(it.minMax()) } }
+
+        val (y_min, y_max) = when(fixed) {
+            true  -> Pair(lineGraph.yFrom, lineGraph.yTo)
+            false -> {
+                Pair(bounds.minY, bounds.maxY)
+            }
+        }
+
+        val parameters = getYParameters(y_min.toDouble(), y_max.toDouble(), timeBasedRange, fixed)
+
+        bounds.minY = parameters.bounds_min
+        bounds.maxY = parameters.bounds_max
+
+        val intervalParameters = Pair(parameters.step_mode, parameters.n_intervals)
+
+        return Pair(bounds, intervalParameters)
+    }
+
+
     private suspend fun getBounds(
```

Comment:
There is this yield call I didn't copy into the new function, not sure what exactly it does and if it has a place in the new code. Besides that, it can be removed.

Reply:
yield and suspend are both related to the use of kotlin co-routines ( https://kotlinlang.org/docs/coroutines-overview.html ) .. I don't think you need the yield here really but I may add it if I notice a slow down. It's just about breaking the work up into chunks so as not to steal too many resources at once and cause frame drops.